server/src/main/java/io/crate/metadata/SystemTable.java

```diff
@@ -397,7 +397,14 @@ public P endObject() {
         }
 
         private <U> void addColumnToParent(Column<T, U> column) {
-            parent.add(new Column<>(column.column, column.type, column.getProperty));
+            if (column instanceof DynamicColumn dynamicColumn) {
+                parent.add(new DynamicColumn<>(
+                    dynamicColumn.column,
+                    dynamicColumn.leafType,
+                    t -> objectIsNull.test(t) ? Map.of() : (Map<String, Object>) dynamicColumn.getProperty.apply(t)));
+            } else {
+                parent.add(new Column<>(column.column, column.type, column.getProperty));
+            }
```

Comment:
What happened was that the DynamicColumn was converted to a Column such that `SystemTable.getReadReference` failed to find the column.